packages/core/src/client.ts

```diff
@@ -549,7 +549,10 @@ export const Client: new (opts: ClientOptions) => Client = function Client(
     fetchSubscriptions: opts.fetchSubscriptions,
     fetchOptions: opts.fetchOptions,
     fetch: opts.fetch,
-    preferGetMethod: opts.preferGetMethod || 'within-url-limit',
+    preferGetMethod:
+      opts.preferGetMethod !== null && opts.preferGetMethod !== undefined
```

Comment:
This change also needs to go into exchange-persisted https://github.com/urql-graphql/urql/blob/main/exchanges/persisted/src/persistedExchange.ts#L141 and a patch changeset for both

```suggestion
      opts.preferGetMethod != null
```